apps/postybirb-ui/src/website-components/bluesky/bluesky-login-view.tsx

```diff
@@ -0,0 +1,147 @@
+import { notifications } from '@mantine/notifications';
+import { BlueskyAccountData, BlueskyOAuthRoutes } from '@postybirb/types';
+import React, { useState } from 'react';
+import websitesApi from '../../api/websites.api';
+import { ExternalLink } from '../../components/external-link/external-link';
+import HttpErrorResponse from '../../models/http-error-response';
+import { LoginComponentProps } from '../../models/login-component-props';
+
+const formId = 'bluesky-login-form';
+
+export default function BlueskyLoginView(
+  props: LoginComponentProps<BlueskyAccountData>,
+): JSX.Element {
+  const { account } = props;
+  const { data, id } = account;
+  const [username, setUsername] = useState(data?.username ?? '');
+  const [password, setPassword] = useState(data?.password ?? '');
+  const [isSubmitting, setSubmitting] = useState<boolean>(false);
+
+  const usernameValidationErrors: React.ReactNode[] = [];
+  if (username.startsWith('@')) {
+    usernameValidationErrors.push(
+      <Trans comment="Bluesky login form">
+        You don't need to input the @. Unless your username{' '}
+        <strong>really</strong> contains it
+      </Trans>,
+    );
+  }
+  if (username && !username.includes('.')) {
+    usernameValidationErrors.push(
+      <Trans comment="Bluesky login form">
+        Be sure that the username is in the format handle.bsky.social. Or if you
+        are using custom domain, make sure to include full username, e.g.
+        domain.ext, handle.domain.ext
+      </Trans>,
+    );
+  }
+
+  return (
+    <form
+      id={formId}
+      onSubmit={(event) => {
+        event.preventDefault();
+        setSubmitting(true);
+        websitesApi
+          .performOAuthStep<BlueskyOAuthRoutes>(id, 'login', {
+            username,
+            password,
+          })
+          .then(({ result }) => {
+            if (result) {
+              notifications.show({
+                title: 'Login success.',
+                message: 'Login success.',
+                color: 'green',
+              });
+              setPassword('');
+            } else {
+              notifications.show({
+                title: 'Login failed.',
+                message: 'Check that handle and password are valid.',
+                color: 'red',
+              });
+            }
+          })
+          .catch(({ error }: { error: HttpErrorResponse }) => {
+            notifications.show({
+              title: (
+                <span>
+                  {error.statusCode} {error.error}
+                </span>
+              ),
+              message: error.message,
+              color: 'red',
+            });
+          })
+          .finally(() => {
+            setSubmitting(false);
+          });
+      }}
+    >
+      <Stack>
+        <TextInput
+          label={<Trans>Username</Trans>}
+          name="username"
+          description={
+            <Trans>Your DID or handle - for example yourname.bsky.social</Trans>
```

Comment:
nit: acronyms should just be spelt out here or removed. I feel kind of bad for translators on these kinds of sentences.